sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureCliCredential.java

```diff
@@ -23,16 +24,20 @@ public class AzureCliCredential implements TokenCredential {
     private static final ClientLogger LOGGER = new ClientLogger(AzureCliCredential.class);
 
     private final IdentityClient identityClient;
+    private final IdentitySyncClient identitySyncClient;
 
     /**
      * Creates an AzureCliSecretCredential with default identity client options.
      * @param tenantId the tenant id of the application
      * @param identityClientOptions the options to configure the identity client
      */
     AzureCliCredential(String tenantId, IdentityClientOptions identityClientOptions) {
-        identityClient = new IdentityClientBuilder()
+        IdentityClientBuilder builder = new IdentityClientBuilder()
             .tenantId(tenantId)
-            .identityClientOptions(identityClientOptions).build();
+            .identityClientOptions(identityClientOptions);
+
+        identityClient = builder.build();
+        identitySyncClient = builder.buildSyncClient();
```

Comment:
We talked about measuring the cost of this and perhaps optimizing it to lazily create a client based on how a particular credential is used. I know we aren't doing it for this PR but can you open an issue to track doing it?

Reply:
Created the issue: https://github.com/Azure/azure-sdk-for-java/issues/31462